lib/que/locker.rb

```diff
@@ -24,9 +24,23 @@ def initialize(queue:, cursor_expiry:, metrics:)
     def with_locked_job
       reset_cursor if cursor_expired?
 
+      job = lock_job
+
+      # Becuase we were using a cursor when we tried to lock this job, if we fail to find
+      # a job it is not necessarily the case that there aren't jobs in the queue. We may
+      # have been excluding candidate jobs due to the cursor that are now due to be
+      # worked.
+      #
+      # We should attempt to lock again after resetting our cursor to make sure we include
+      # jobs that were excluded in the first attempt.
+      if job.nil? && @cursor
```

Comment:
@cursor is 0 if there is no cursor, right?
If that's the case, @cursor is always true, and we'll go into this if no matter what.

So, this is either a bug, or we're just redundantly checking `@cursor`, and we'll reset and reload anyway even if there was no cursor.